packages/react/src/components/dropzone.tsx

```diff
@@ -149,13 +149,17 @@ export function UploadDropzone<
       if (document.activeElement !== rootRef.current) return;
 
       const pastedFiles = getFilesFromClipboardEvent(event);
-      if (!pastedFiles) return;
+      if (!pastedFiles?.length) return;
 
-      setFiles((prev) => [...prev, ...pastedFiles]);
+      let filesToUpload = pastedFiles;
+      setFiles((prev) => {
+        filesToUpload = [...prev, ...pastedFiles];
```

Comment:
idk if this is cursed or good...

Reply:
How is this different from before....? 😅

Reply:
State update cycle

Reply:
🥲